inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -827,3 +829,78 @@ def extract_authors_from_pdf(obj, eng):
                 grobid_authors_count
             )
     return obj
+
+
+def prepare_collaboration_multi_search(collaborations):
+    multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    backup_multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    for collaboration in collaborations:
+        full_collaboration_string = collaboration.get('value', '')
+        normalized_collaboration_string = normalize_collaboration_name(full_collaboration_string)
+        if collaboration.get('record'):
+            # Add dummy search so multisearch will stay in sync with collaborations
+            multi_search = multi_search.add(Search().query().source(False))
+            backup_multi_search = backup_multi_search.add(Search().query().source(False))
+            continue
+
+        multi_search = multi_search.add(build_standard_collaboration_search(normalized_collaboration_string))
+        backup_multi_search = backup_multi_search.add(build_subgroup_collaboration_search(normalized_collaboration_string))
```

Comment:
```suggestion
        backup_multi_search = subgroup_multi_search.add(build_subgroup_collaboration_search(normalized_collaboration_string))
```
but why do you need two multi-searches? If you already have a multi-search, couldn't you add both searches here to the same multi-search?

Reply:
It's easier to iterate through results. I can do all of them in one multi_search, but it will be a little hackish.